runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -460,14 +418,19 @@ public void reconfigureSequencerServers(Layout originalLayout, Layout newLayout,
                         .equals(newLayout.getPrimarySequencer())) {
                     Layout.LayoutSegment latestSegment = newLayout.getSegments()
                             .get(newLayout.getSegments().size() - 1);
-                    maxTokenRequested = getMaxGlobalTail(newLayout, latestSegment);
+                    maxTokenRequested = getMaxGlobalTail(newLayout, latestSegment, runtime);
 
                     FastObjectLoader fastObjectLoader = new FastObjectLoader(runtime);
                     fastObjectLoader.setRecoverSequencerMode(true);
                     fastObjectLoader.setLoadInCache(false);
 
                     // FastSMRLoader sets the logHead based on trim mark.
-                    fastObjectLoader.setLogTail(maxTokenRequested);
+
+                    // This is specifically set to 0 if maxTokenRequested is -1 because we do not
+                    // want the FastLoader to calculate its own tail based on the Runtime's layout
+                    // and want to calculate it on the newLayout achieved by consensus.
+                    long tail = maxTokenRequested == Address.NON_ADDRESS ? 0L : maxTokenRequested;
```

Comment:
You would want to remove this right. The FastLoader has a default of -6 now to support this. Also update the comment accordingly.
And also initialize `maxTokenRequested` to -1 instead?